### PR TITLE
Support a few corner cases for nvFuser executor

### PR DESCRIPTION
**Pull request by IvanYashchuk**
This PR adds asserts to the `nvfuser_execute` function for the cases that do not work. Fallback to eager is used in those cases.

### CODE REVIEW

File: test/test_prims.py [hunk 1]
```diff
@@ -157,6 +157,76 @@ def test_nvfuser_impl_is_used(self, device):
...
+        a = torch.randn(3, 3, device=device)
+        b = torch.randn(3, 3, device=device)
+
+        def func(b):
+            return a + b
+
+        with TorchRefsNvfuserCapabilityMode():
+            gm = make_fx(func)(b)
+
+        with self.assertRaisesRegex(AssertionError, "not supported yet"):
```
Comment (jjsjann123) [inline test/test_prims.py:222]: nitpick: we can use more explicit error message for the assert.
`Constant tensors that are saved in the graph and used as arguments are not supported yet`

File: torch/_prims/nvfuser_executor.py [hunk 2]
```diff
@@ -68,6 +84,21 @@ def make_nvfuser_fusion(gm: GraphModule, *nv_args_templates):
...
+        filter(lambda n: n.op == "call_function", gm.graph.nodes)
+    )
+    assert len(graph_input_nodes) == len(
+        nv_args_templates
+    ), "Number of placeholder nodes in the graph must match number of args"
+    assert len(nv_args_templates) > 0, "There must be at least one argument"
+    assert (
+        len(call_function_nodes) > 0
+    ), "Graph must contain at least one call_function node"
+    assert not _any_get_attr_used(
```
Comment (jjsjann123) [inline torch/_prims/nvfuser_executor.py:98]: Any reason that we are explicitly asserting here?

I remember seeing some issues with aot mistakenly converting things to constant tensor, but I don't know if those are always just wrong.

**Review by jjsjann123 — APPROVED** [review]
LGTM with a few questions.

File: torch/_prims/nvfuser_executor.py (hunk 2 above)
Comment (IvanYashchuk) [reply torch/_prims/nvfuser_executor.py:98]: Without this assertion the error comes from the `fusion.execute` call and it's less readable.

**Review by ngimel — APPROVED** [review]

File: test/test_prims.py (hunk 1 above)
Comment (IvanYashchuk) [reply test/test_prims.py:222]: I was usually using substring to match the error messages in tests. I'll be more careful trying to use a more complete message to be checked next time.

File: test/test_prims.py [hunk 3]
```diff
@@ -157,6 +157,76 @@ def test_nvfuser_impl_is_used(self, device):
...
+            execute(gm, b, executor="strictly_nvfuser")
+
+        # Should pass with partitioned executor
+        gm.graph.print_tabular()
```
Comment (IvanYashchuk) [inline test/test_prims.py:226]: It shouldn't be here.
```suggestion
```

**Comment by huydhn** [comment]
Please rebase into master to get rid of the build failures. They are from my very recent commit https://github.com/pytorch/pytorch/commit/cb4421b19cf7aa3a1a6cfcc7e0677f2b2ba0a9b6 and the new workflow file hasn't made it to viable/strict yet.  Thank you for your understanding.

**Pull request closed without merging**